### PR TITLE
Prep the `StudentSelector`, `GradingToolbar` components for UI enhancements

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR updates the `StudentSelector` component and makes some small adjustments to the `GradingToolbar` in advance of some UI enhancements in this area.

It:

* Converts `StudentSelector` to TSX
* Updates the `@hypothesis/frontend-shared` package such that the `Select` component is available
* Updates shared components in `StudentSelector`, including the use of `Select` and simplifies component use/layout
* Tightens and lightens some text in `GradingToolbar` to reduce vertical space used and to de-emphasize the course name a tad

User-facing changes:

* Text weight and color of assignment name, course name at left side of `GradingBar`
* Shorter overall height of `GradingBar`[^1]
* Padding and proportional icon sizes in the student-selector drop-down and adjacent `IconButton`s. These small adjustments bring these UI bits more into line with our common UI patterns.

Before:

image

After[^2]:

image

(Despite apparent differences, the select (drop-down) field is the same width in both screenshots).

[^1]: Don't worry, I did read the brief. I know we're going to want to make this bit of interface yet shorter for situations in which the grading elements are not relevant. This is just a starting step.
[^2]: I am having some woes with my local Via this afternoon which prevent me from getting a localhost assignment to a place where there are students who have made annotations, thus preventing me from showing a non-empty select list here.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/StudentSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          </span>
+        ) : (
+          <span>{`${students.length} Students`}</span>
+        )}
+      </label>
+      <div>
+        <InputGroup>
+          <IconButton
+            icon={ArrowLeftIcon}
+            title="previous student"
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/StudentSelector.tsx:80]:
```suggestion
            title="Previous student"
```

These titles are user visible in the form of tooltips.

File: lms/static/scripts/frontend_apps/components/StudentSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                selected={selectedStudentIndex === i}
+                value={i}
+              >
+                {student.displayName}
+              </option>
+            ))}
+          </Select>
+          <IconButton
+            icon={ArrowRightIcon}
+            title="next student"
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/StudentSelector.tsx:111]:
```suggestion
            title="Next student"
```

File: lms/static/scripts/frontend_apps/components/StudentSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  /** Callback invoked when the selected student changes */
+  onSelectStudent: (index: number) => void;
+  /** Index of selected student in `students` or -1 if no student is selected */
+  selectedStudentIndex: number;
+  /** Ordered list of students to display in the drop-down */
+  students: Student[];
+};
+
+/**
+ * A student navigation tool which shows a student selection list and a previous and next button.
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/StudentSelector.tsx:24]: I know this was copied from the previous code, but I think the wording here could be refined. Maybe something like "Control which allows selecting a student from a list."

**Review by robertknight — APPROVED** [review]
LGTM with a few minor comments.

File: lms/static/scripts/frontend_apps/components/StudentSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      >
+        {selectedStudentIndex >= 0 ? (
+          <span>
+            Student {`${selectedStudentIndex + 1} of ${students.length}`}
+          </span>
+        ) : (
+          <span>{`${students.length} Students`}</span>
+        )}
+      </label>
+      <div>
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/StudentSelector.tsx:76]: Needed for flex layout as intended (as an immediate child to a flex container). I'll see if I can clarify with a comment.

File: lms/static/scripts/frontend_apps/components/StudentSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  const hasPrevView = selectedStudentIndex >= 0;
+
+  /**
+   * Select the next student index in the list.
+   */
+  const onNextView = () => {
+    onSelectStudent(selectedStudentIndex + 1);
+  };
+  /**
+   * Select the previous student index in the list. The 0 index
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/StudentSelector.tsx:45]: Yeah, I'm just gonna strip the JSDoc comments for both of these callbacks, as they are self-documenting.

**Pull request merged**